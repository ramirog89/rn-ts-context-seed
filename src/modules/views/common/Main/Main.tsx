import React, { useEffect, useState } from 'react';
import { SafeAreaView, StatusBar, ActivityIndicator } from 'react-native';
import { SafeAreaProvider } from 'react-native-safe-area-context';

import { useAuth } from '../../../context/auth';

import Router from '../Router';
import ToastList from '../ToastList';
import ErrorBoundary from '../ErrorBoundary';

const Main = () => {
  const { state, recoverSession } = useAuth();
  const [isLoading, setLoading] = useState(true);

  useEffect(() => {
    recoverSession();
  }, []);

  useEffect(() => {
    if (state.sessionChecked) {
      setLoading(false);
    }
  }, [state.sessionChecked]);

  if (isLoading) {
    return (
      <ActivityIndicator
        style={{ alignSelf: 'center', flex: 1 }}
        size="large"
        color="#777777"
        animating
      />
    );
  }

  return (
    <SafeAreaProvider>
      <StatusBar translucent={true} backgroundColor="transparent" barStyle="light-content" />
      <SafeAreaView style={{ flex: 1 }}>
        <ErrorBoundary>
          <Router />
        </ErrorBoundary>
        <ToastList />
      </SafeAreaView>
    </SafeAreaProvider>
  );
}

export default Main;
