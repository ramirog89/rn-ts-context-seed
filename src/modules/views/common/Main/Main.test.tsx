import React from 'react';

import { render } from '../../../../test/render';

import Main from './Main';

jest.useFakeTimers();

describe('Main', () => {
  it('should render', () => {
    const { toJSON } = render(<Main />);
    expect(toJSON()).toMatchSnapshot();
  });
});
