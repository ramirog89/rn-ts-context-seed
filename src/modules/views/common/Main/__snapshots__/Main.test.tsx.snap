// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should render 1`] = `
<ActivityIndicator
  animating={true}
  color="#777777"
  hidesWhenStopped={true}
  size="large"
  style={
    Object {
      "alignSelf": "center",
      "flex": 1,
    }
  }
/>
`;
